src/include/miopen/solver/implicitgemm_util.hpp

```diff
@@ -607,6 +607,11 @@ template <typename T>
 T gcd(T x, T y)
 {
     assert(!(x == 0 && y == 0));
+    // for clang
+    if(x == 0 && y == 0)
+    {
+        return 1;
```

Comment:
gcd is undefined when `(x == 0 && y == 0)` so it is incorrect to return 1.

Can you please clarify why this is necessary? IIRC we were sure that `(x == 0 && y == 0)` would never happen, and therefore decided to remove run-time checks from release builds for the sake of performance. Finally we decided to add assert so debug builds can catch the problem (see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/243). So far, clang hasn't complained.

If we really need to handle this case in release builds, then let's replace assert with MIOPEN_THROW.

Reply:
There is an issue with clang. When clang is performing branch exploration it is certain that this function can return zero (because x and y are both zero). This obviously cannot happen, because of the assert. Possibly because of the recursive logic here, the compiler abbreviates its check and assumes this function returns a zero, in which case there is a divide by 0 exception in a caller to this function. This unreachable statement here resolves the issue for clang.

Reply:
I see. Do you mean clang-tidy checks or warning during compilation with clang?

Reply:
It was a make check error.

Reply:
I'll add this.

Reply:
It seems like cppcheck-suppress is not necessary anymore. Resolved.